src/data/finsupp/defs.lean

```diff
@@ -84,22 +84,76 @@ open_locale classical big_operators
 
 variables {α β γ ι M M' N P G H R S : Type*}
 
-/-- `finsupp α M`, denoted `α →₀ M`, is the type of functions `f : α → M` such that
-  `f x = 0` for all but finitely many `x`. -/
-structure finsupp (α : Type*) (M : Type*) [has_zero M] :=
+/-- `finsupp' α M v` is the type of functions `f : α → M` such that
+  `f x = v` for all but finitely many `x`. -/
+structure finsupp' (α : Type*) (M : Type*) (v : M) :=
 (support            : finset α)
 (to_fun             : α → M)
-(mem_support_to_fun : ∀a, a ∈ support ↔ to_fun a ≠ 0)
+(mem_support_to_fun : ∀a, a ∈ support ↔ to_fun a ≠ v)
+
+/-- `finsupp α M`, denoted `α →₀ M`, is the type of functions `f : α → M` such that
+  `f x = 0` for all but finitely many `x`. -/
+def finsupp (α : Type*) (M : Type*) [has_zero M] :=
```

Comment:
Making this `@[reducible]` will make some duplication below unnecessary

Reply:
What exactly seems unnecessary? I think I can remove the `fun_like` and `has_coe_to_fun` instances, but is there anything else you were thinking of?